tools/adventure-pack/goodies/typescript/Iterator.prototype.find/index.test.ts

```diff
@@ -0,0 +1,58 @@
+import { describe, expect, it } from "@jest/globals";
+
+import "./index";
+
+describe("Iterator.prototype.find", () => {
+  it("returns the first element that satisfies the callback", () => {
+    const array = [5, 2, 1, 7, -2];
+    expect(array.values().find((element) => element % 2 === 0)).toBe(2);
+    expect(array.values().find((element) => element > 5)).toBe(7);
+    expect(array.values().find((element) => element > 0)).toBe(5);
+  });
+
+  it("returns undefined if no element satisfies the callback", () => {
+    const array = [-8, -3, -9, -2];
+    expect(array.values().find((element) => element > 0)).toBeUndefined();
+  });
+
+  it("returns undefined for an empty iterator", () => {
+    expect([].values().find(() => true)).toBeUndefined();
+  });
+
+  it("can test Map entries()", () => {
+    const map = new Map([
+      ["aPpLes", { type: "fruit", price: 1.99 }],
+      ["CHICKEN", { type: "meat", price: 10.99 }],
+      ["oranges", { type: "fruit", price: 3.99 }],
+      ["StEaK", { type: "meat", price: 8.99 }],
+      ["GRAPES", { type: "fruit", price: 2.99 }],
+    ]);
+
+    expect(
+      map
+        .entries()
+        .find(
+          ([key, value]) =>
+            key === key.toUpperCase() &&
+            value.type === "fruit" &&
+            value.price < 3.0,
+        ),
+    ).toEqual(["GRAPES", { type: "fruit", price: 2.99 }]);
+  });
+
+  it("can test Set values()", () => {
+    const set = new Set([2, 2, -3, -3, 4, -5, 5, -6]);
+    expect(set.values().find((element) => element > 4)).toBe(5);
```

Comment:
To make the test more interesting, let's add another element that satisfies the requirement. Then we are also testing that `.find` finds the first one that matches!

Reply:
ack!